## Linear Regression Modelling

In [1]:
import pandas as pd
df= pd.read_excel("C:\\Users\\heeckim\\Desktop\\work\\12_python\\pyharry\\resources\\lrdata.xlsx")

In [2]:
import statsmodels.api as sm
X = df['performance_rating']
Y = df['percentage_salary_hike']
X = sm.add_constant(X)
model1=sm.OLS(Y,X) # Ordinary Least Square Model  사용
result=model1.fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                              OLS Regression Results                              
==================================================================================
Dep. Variable:     percentage_salary_hike   R-squared:                       0.598
Model:                                OLS   Adj. R-squared:                  0.598
Method:                     Least Squares   F-statistic:                     2187.
Date:                    Mon, 21 Feb 2022   Prob (F-statistic):          4.27e-293
Time:                            15:26:25   Log-Likelihood:                -3322.1
No. Observations:                    1470   AIC:                             6648.
Df Residuals:                        1468   BIC:                             6659.
Df Model:                               1                                         
Covariance Type:                nonrobust                                         
======================================================================================
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
const                 -9.5358      0.533    -17.905      0.000     -10.580      -8.491
performance_rating     7.8463      0.168     46.767      0.000       7.517       8.175
==============================================================================
Omnibus:                      126.254   Durbin-Watson:                   2.106
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              111.830
Skew:                           0.602   Prob(JB):                     5.20e-25
Kurtosis:                       2.388   Cond. No.                         30.7
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

* const/coef = -9.5358 = performance_rating이 0일때의 추정치
* performance_raiting/coef = 7.8463 = 슬로프, 퍼포먼스레이팅이 1증가할때마다 셀러리 7.8463 증가
* p(>|t|): 슬로프의 유의미성, 0.05보다 작으면 통계적으로 유의미하다
* salary hike에 대하여 performance rating로 예측하는 회귀분석을 실시한 결과, 이 회귀모형은 통계적으로 유의미하였다(F(Df Model,Df Residual) = F-statistics, p < 0.05) = (F(1, 1468) = 2187, p < 0.05>)
* performance rating의 slpoe는 7.8463으로 salary hike에 유의미한 예측 변인인 것으로 나타났다(t(1468)=46.767, p < 0.05)

In [4]:
df1=df.drop(['percentage_salary_hike'],axis=1) # salary_hike를 제외한 모든 independent variables의 salary_hike와 연관성을 보기위하여 준비작업

In [5]:
import statsmodels.api as sm
X = df1
Y = df['percentage_salary_hike']
X = sm.add_constant(X)
model1=sm.OLS(Y,X) # Ordinary Least Square Model  사용
result=model1.fit()
result.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                              OLS Regression Results                              
==================================================================================
Dep. Variable:     percentage_salary_hike   R-squared:                       0.602
Model:                                OLS   Adj. R-squared:                  0.598
Method:                     Least Squares   F-statistic:                     137.3
Date:                    Mon, 21 Feb 2022   Prob (F-statistic):          1.50e-276
Time:                            16:05:09   Log-Likelihood:                -3315.6
No. Observations:                    1470   AIC:                             6665.
Df Residuals:                        1453   BIC:                             6755.
Df Model:                              16                                         
Covariance Type:                nonrobust                                         
==============================================================================================
                                 coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------
const                         -9.6431      0.639    -15.092      0.000     -10.896      -8.390
age                            0.0129      0.009      1.381      0.167      -0.005       0.031
job_level                     -0.1126      0.186     -0.605      0.545      -0.478       0.252
monthly_income              3.785e-05   4.24e-05      0.892      0.373   -4.54e-05       0.000
performance_rating             7.8529      0.168     46.635      0.000       7.523       8.183
total_working_years           -0.0174      0.017     -1.044      0.297      -0.050       0.015
training_times_last_year       0.0168      0.047      0.354      0.723      -0.076       0.110
years_since_last_promotion    -0.0264      0.024     -1.086      0.278      -0.074       0.021
years_with_curr_manager        0.0024      0.028      0.086      0.931      -0.053       0.058
years_at_company              -0.0136      0.021     -0.660      0.510      -0.054       0.027
years_in_current_role          0.0044      0.027      0.160      0.873      -0.049       0.058
dummy_sales                    0.0213      0.137      0.156      0.876      -0.248       0.290
dummy_HR                      -0.4001      0.304     -1.317      0.188      -0.996       0.196
dummy_female                  -0.0997      0.124     -0.802      0.423      -0.344       0.144
dummy_divorced                -0.1617      0.157     -1.030      0.303      -0.470       0.146
dummy_single                  -0.0688      0.141     -0.487      0.627      -0.346       0.208
dummy_yes                     -0.0698      0.135     -0.516      0.606      -0.335       0.196
==============================================================================
Omnibus:                      122.725   Durbin-Watson:                   2.115
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              110.365
Skew:                           0.601   Prob(JB):                     1.08e-24
Kurtosis:                       2.401   Cond. No.                     8.70e+04
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 8.7e+04. This might indicate that there are
strong multicollinearity or other numerical problems.
"""